materialize-boilerplate/apply.go

```diff
@@ -0,0 +1,229 @@
+package boilerplate
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	maxConcurrentUpdateActions = 5
+)
+
+// ActionApplyFn is a callback that will be executed to carry out some action to achieve a change to
+// a materialized resource.
+type ActionApplyFn func(context.Context) error
+
+// BindingUpdate is a distilled representation of the typical kinds of changes a destination system
+// will care about in response to a new binding or change to an existing binding.
+type BindingUpdate struct {
+	// NewProjections is the projections of a proposed materialization spec that are included in the
```

Comment:
:+1: on the grammar fix

It doesn't take an EndpointField because there is not a materialized "column" in the destination since its a new projection, hence there is no existing field information to create an EndpointField from. Indeed, the Projection must be used to create a new column in the destination system. I'll extend the comment a bit to describe that.

Reply:
Right, I suppose I was thinking "why can't the caller of Applier do more of the heavy-lifting for dialect mapping", but only later realized "because dialects are a SQL-ism only".

Then I wondered whether there was any helpful middle-ground between "here's a Projection, figure it out" vs the nice unpacking work we do for SQL, but nothing obvious came to me.